test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpLogExporterTests.cs

```diff
@@ -27,6 +27,139 @@ public class OtlpLogExporterTests : Http2UnencryptedSupportTests
+
+        if (requestLoggerProviderDirectly)
+        {
+            var provider = serviceProvider.GetRequiredService<LoggerProvider>();
+            Assert.NotNull(provider);
+        }
+        else
+        {
+            var factory = serviceProvider.GetRequiredService<ILoggerFactory>();
+            Assert.NotNull(factory);
+        }
+    }
+
+    [Fact]
+    public void AddOtlpExporterWithNamedOptions()
+    {
+        int defaultExporterOptionsConfigureOptionsInvocations = 0;
+        int namedExporterOptionsConfigureOptionsInvocations = 0;
+
+        using var loggerProvider = Sdk.CreateLoggerProviderBuilder()
+            .ConfigureServices(services =>
+            {
+                services.Configure<OtlpExporterOptions>(o => defaultExporterOptionsConfigureOptionsInvocations++);
+                services.Configure<LogRecordExportProcessorOptions>(o => defaultExporterOptionsConfigureOptionsInvocations++);
+
+                services.Configure<OtlpExporterOptions>("Exporter2", o => namedExporterOptionsConfigureOptionsInvocations++);
+                services.Configure<LogRecordExportProcessorOptions>("Exporter2", o => namedExporterOptionsConfigureOptionsInvocations++);
+
+                services.Configure<OtlpExporterOptions>("Exporter3", o => namedExporterOptionsConfigureOptionsInvocations++);
+                services.Configure<LogRecordExportProcessorOptions>("Exporter3", o => namedExporterOptionsConfigureOptionsInvocations++);
+            })
+            .AddOtlpExporter()
+            .AddOtlpExporter("Exporter2", o => { })
+            .AddOtlpExporter("Exporter3", o => { })
+            .Build();
+
+        Assert.Equal(2, defaultExporterOptionsConfigureOptionsInvocations);
+        Assert.Equal(4, namedExporterOptionsConfigureOptionsInvocations);
+    }
+
+    [Fact]
+    public void UserHttpFactoryCalledWhenUsingHttpProtobuf()
+    {
+        OtlpExporterOptions options = new OtlpExporterOptions();
+
+        var defaultFactory = options.HttpClientFactory;
+
+        int invocations = 0;
+        options.Protocol = OtlpExportProtocol.HttpProtobuf;
+        options.HttpClientFactory = () =>
+        {
+            invocations++;
+            return defaultFactory();
+        };
+
+        using (var exporter = new OtlpLogExporter(options))
+        {
+            Assert.Equal(1, invocations);
+        }
+
+        using (var provider = Sdk.CreateLoggerProviderBuilder()
+            .AddOtlpExporter(o =>
+            {
+                o.Protocol = OtlpExportProtocol.HttpProtobuf;
+                o.HttpClientFactory = options.HttpClientFactory;
+            })
+            .Build())
+        {
+            Assert.Equal(2, invocations);
+        }
+
+        options.HttpClientFactory = null;
+        Assert.Throws<InvalidOperationException>(() =>
+        {
+            using var exporter = new OtlpLogExporter(options);
+        });
+
+        options.HttpClientFactory = () => null;
```

Comment:
why is this twice?

Reply:
I took a look, doesn't seem to be a reason for it. Probably copy/paste fail. I removed the extra block.